index.js

```diff
@@ -1,48 +1,63 @@
-/**
- * jstransformer-absurd <https://github.com/tunnckoCore/jstransformer-absurd>
- *
- * Copyright (c) 2015 Charlike Mike Reagent, contributors.
- * Released under the MIT license.
- */
-
 'use strict';
 
-var fs = require('fs');
-var path = require('path');
-var absurd = require('absurd')();
-var transformer = require('jstransformer');
-var postcss = transformer(require('jstransformer-postcss'));
-var nested = require('postcss-nested');
+var Absurd = require('absurd');
+var Promise = require('promise');
+var merge = require('merge');
 
 exports.name = 'absurd';
-exports.inputFormats = ['absurd', 'absurdjs', 'js'];
-exports.outputFormat = 'css';
+exports.inputFormats = ['absurd', 'absurdjs'];
+// Absurd can output results in multiple formats. We output with "absurd"
+// to refrain from making an asumption on how it will be used.
+exports.outputFormat = 'absurd';
 
-exports.render = function _render(api, options) {
-  var isFunction = typeof api === 'function';
+/**
+ * Build an Absurd object from the given input, options and locals.
+ */
+var constructAbsurd = function (input, options, locals) {
+  // Build a base Absurd object.
+  var absurd = Absurd();
+
+  // Retrieve the options.
+  options = merge(options || {}, locals || {});
 
-  if (!isFunction && typeof api !== 'object') {
-    throw new TypeError('jstransformer-absurd expects object or function');
+  // Check if we are to morph the object.
+  if (options.morph) {
+    absurd.morph(options.morph);
   }
-  if (isFunction) {
-    api = api(absurd, options);
+
+  // Process the input for the object.
+  if (typeof input == 'string' || input instanceof String) {
```

Comment:
Use duck typing when possible.

Also, isn't `input` not allowed to be anything but a string?